core/domain/state_domain.py

```diff
@@ -1357,6 +1464,78 @@ def validate(self, rule_params_list, exp_param_specs_dict):
                 # by the parameter object in order to be valid.
                 param_obj.normalize(param_value)
 
+    @staticmethod
+    def get_all_html_fields_in_rule_spec(rule_spec_dict):
+        """Checks for HTML in a rule_spec dict and returns all the information
+        about fields having HTML in them.
+
+        Args:
+            rule_spec_dict: dict. The dict representation of RuleSpec object.
+
+        Returns:
+            dict. The dict having the information about all the fields which
+            have HTML.
+        """
+        field_check_dict = {
```

Comment:
Alas, I think what you have here is even messier than the previous iteration.

Let's take it back to what it was before. I take your point about convert_html_in_rule_spec() being used in migrating draft_change_list in which it has no reference to interaction type. This sounds like it needs a refactor of how we handle change lists which is broader than the scope of this PR. Could you therefore please do the following:

(a) file an issue for including the interaction ID with the change dict when updating an interaction's rules (or anything else that depends on the interaction ID) -- reply to this comment with the issue number when you've done that. Please try to make the issue as clear and descriptive as possible, with regards to what the problem is.

(b) Add a comment in the places you're duplicating the selection logic, saying "NOTE TO DEVELOPERS: This logic needs to be kept in sync with [these other places in the code -- insert relevant places here]. We cannot make this declarative until issue #XXX has been fixed, because YYY." and point to the issue in (a).

Thanks!

Reply:
done, created the issue: #9413